crates/ethereum-rpc/src/trace.rs

```diff
@@ -275,6 +333,116 @@ fn get_traces_with_requested_tracer_and_config(
+                value: call_frame.value.unwrap_or_default(),
+                creation_method: match call_type {
+                    "create" => CreationMethod::Create,
+                    "create2" => CreationMethod::Create2,
+                    &_ => {
+                        return Err(EthApiError::Unsupported("Unsupported call type"));
+                    }
+                },
+            }),
+            error: call_frame.error,
+            result: Some(TraceOutput::Create(CreateOutput {
+                address: call_frame.to.unwrap_or_default(),
+                code: call_frame.output.unwrap_or_default(),
+                gas_used: call_frame.gas_used.saturating_to(),
+            })),
+            subtraces: call_frame.calls.len(),
+            trace_address: vec![],
+        },
+        "selfdestruct" => TransactionTrace {
+            action: Action::Selfdestruct(SelfdestructAction {
+                address: call_frame.from,
+                balance: call_frame.value.unwrap_or_default(),
+                refund_address: call_frame.to.unwrap_or_default(),
+            }),
+            error: call_frame.error,
+            result: Some(TraceOutput::Create(CreateOutput {
+                address: call_frame.to.unwrap_or_default(),
+                code: call_frame.output.unwrap_or_default(),
+                gas_used: call_frame.gas_used.saturating_to(),
+            })),
+            subtraces: call_frame.calls.len(),
+            trace_address: vec![],
+        },
+        "call" | "staticcall" | "callcode" | "delegatecall" => TransactionTrace {
+            action: Action::Call(CallAction {
+                from: call_frame.from,
+                call_type: match call_type {
+                    "call" => CallType::Call,
+                    "staticcall" => CallType::StaticCall,
+                    "callcode" => CallType::CallCode,
+                    "delegatecall" => CallType::DelegateCall,
+                    &_ => {
+                        return Err(EthApiError::Unsupported("Unsupported call type"));
+                    }
+                },
+                gas: call_frame.gas.saturating_to(),
+                input: call_frame.input,
+                to: call_frame.to.unwrap_or_default(),
+                value: call_frame.value.unwrap_or_default(),
+            }),
+            error: call_frame.error,
+            result: Some(TraceOutput::Call(CallOutput {
+                gas_used: call_frame.gas_used.saturating_to(),
+                output: call_frame.output.unwrap_or_default(),
+            })),
+            subtraces: call_frame.calls.len(),
+            trace_address: vec![],
+        },
+        _ => {
+            return Err(EthApiError::Unsupported("Unsupported call frame"));
+        }
+    };
+
+    let frame = LocalizedTransactionTrace {
+        trace,
+        block_hash,
+        block_number,
+        transaction_hash: tx_hash,
+        transaction_position: tx_index.map(|i| i as u64),
+    };
```

Comment:
You also need to assign the trace_address to the top level trace otherwise it's always empty

```suggestion
    let mut frame = LocalizedTransactionTrace {
        trace,
        block_hash,
        block_number,
        transaction_hash: tx_hash,
        transaction_position: tx_index.map(|i| i as u64),
    };
    
    frame.trace.trace_address = trace_address.clone();
```

Reply:
https://github.com/chainwayxyz/citrea/pull/2386/commits/dc7de77b70863f38ae4d2760f7d35f2c59c9d7e0